### DISCUSSION TITLE
Adding an "FLoad" dynamic load causing error

### DISCUSSION THREAD

**Discussion by MaKla89** [CURENT/andes#442] [Modeling] [answered]
For a very small test-grid (4 bus, 1 slack, 1 synGen, 2 lines, 2 transformers, ONE load) I tried to go from static load (PQ) to dynamic load (FLoad). Yet, even if I keep the FLoad inactive (u = 0), running TDS raises this error:

`ValueError: All arrays must be of the same length`

The following screenshot contains the FLoad-settings and the Traceback:
![image](https://user-images.githubusercontent.com/33317175/233328945-7e219985-5444-41fb-a304-f4ceeb4afe52.png)

As there is very little to read in the FLoad-documentation, I am not sure whether I'm 100% blind or whether something is missing :/ 

When I delete Row 1 of the xlsx file, then the model is running fine again (obviously without the dynamic load, but it works).

**Comment by jinningwang** [comment]
Hi @MaKla89, it's great to see you here again! It seems like there might be a bug. Would you mind sharing the relevant case file and minimal code with me? I'd be happy to take a look when I have some time.

**Reply by cuihantao** [reply]
Looks like `cases/ieee14/ieee14_fload.json` can replicate it.

It was broken somewhere between version 1.5.0 and 1.6.0. Doing a git bisect can identify the commit with issue.

**Reply by MaKla89** [reply]
> Hi @MaKla89, it's great to see you here again! It seems like there might be a bug. Would you mind sharing the relevant case file and minimal code with me? I'd be happy to take a look when I have some time.

Sure thing, thanks! :) Here you go, that system is so small, that it can directly be used as a minimal case:
[v_coll_sys.xlsx](https://github.com/CURENT/andes/files/11285567/v_coll_sys.xlsx)

Here the minimal-code:
[FLoad problem.txt](https://github.com/CURENT/andes/files/11285658/FLoad.problem.txt)

**Comment by jinningwang** [comment]
The version test for 1.5.2 failed, but the test was successful for version 1.4.4.

Minimal code:
```python
ss = andes.load(andes.get_case('ieee14/ieee14_fload.json'))
ss.PFlow.run()
ss.TDS.init()
```

**Reply by MaKla89** [reply]
Hmm, do you have an idea whether this will be an easy fix or whether I should rather use version 1.4.4 instead for now? :)

**Reply by jinningwang** [reply]
It seems to be a quick fix and I'm working on this. Expect to solve this in 1-2 days.

**Reply by MaKla89** [reply]
Excellent news, thanks for the great support and communication :)

**Reply by jinningwang** [reply]
Using 1.4.4 can be a possible option before this was fixed, I don't have a clear timeline to fix this.

**Reply by jinningwang** [reply, marked as answer]
Update: Dr. Cui has fixed this, see https://github.com/CURENT/andes/issues/444?notification_referrer_id=NT_kwDOBLjkvbM2MjMyMTU3NzAyOjc5MjI2MDQ1

**Reply by MaKla89** [reply]
Perfect, I'll try it right away and mark this as the answer to my question :)

**Comment by jinningwang** [comment]
@cuihantao Hantao, I'm having trouble figuring out how to fix this error. Would you be able to take a look at it when you have some free time?

## Error

ANDES version: 1.5.1.post2+ge8465ce0, case: `ieee14_fload.json`

The error occurred in method ``ModelCache.refresh`` in ``model.py``. The error was caused by the bus value being none by ``ss.FLoad.as_dict(vin=True)``. Output results are shown below:

```python
ss.FLoad.as_dict(vin=False)

{'uid': array([0]),
...
 'bus': [2]}

ss.FLoad.as_dict(vin=True)

{'uid': array([0]),
...
 'bus': array([], dtype=float64)}
```

## First bad commit

After inspection, the issue first occurred in commit e8465ce0264304fee945b8968db53fa5206849e2. The procedures to identify it are listed below:

1. Use `git bisect` to locate the first known issue commit.
1. Start the git bisect by running `git bisect`.
1. Mark the first known good commit 9c2a5c84 using `git bisect good 9c2a5c84`.
1. Mark the first known bad commit f417cf2e using `git bisect bad f417cf2e`.
1. Perform the bisection method and test run, and mark each as good or bad, by running the following commands:
```
git checkout <commit>
pip install -e .
andes prep
andes st
andes run <andes_path>/andes/cases/ieee14/ieee14_fload.json -r tds
```

Finally, the first bad commit can be identified.
